### PR TITLE
Add test data for baseline/dimension addition, make schema refactor a…

### LINKED ISSUE #366: Add new dimension element (as a child of the baseline element)
Add a new element `iati-activities/iati-activity/result/indicator/baseline/dimension` with definition:

> A category used for disaggregating the result by gender, age, etc.

The element will have an occurrence rule of `0..*` and will replicate the existing `iati-activities/iati-activity/result/indicator/period/actual/dimension` element.

This will also add the following sub-attributes with definitions:

- `iati-activities/iati-activity/result/indicator/baseline/dimension/@name` - occurs `0..1`, type `xsd:string`
> Freetext description of a category being disaggregated.

- `iati-activities/iati-activity/result/indicator/baseline/dimension/@value` - occurs `0..1`, type `xsd:string`
> Description of the value being disaggregated.

**Tasks:**

- [x] Add tests
- [x] Add `iati-activities/iati-activity/result/indicator/baseline/dimension`
- [x] Add `iati-activities/iati-activity/result/indicator/baseline/dimension/@name`
- [x] Add `iati-activities/iati-activity/result/indicator/baseline/dimension/@value`

### PR DESCRIPTION
**Pull request by allthatilk**
…nd addition.

Fixes #366

### CODE REVIEW

File: iati-activities-schema.xsd [hunk 1]
```diff
@@ -2867,5 +2814,33 @@
     </xsd:annotation>
   </xsd:attribute>
 
+  <xsd:element name="dimension">
+    <xsd:annotation>
+      <xsd:documentation xml:lang="en">
+        A category used for disaggregating the result by gender,
+        age, etc.
```
Comment (hayfield) [inline iati-activities-schema.xsd]: Mid-sentence hard-wrapping

File: tests/should-pass/iati-activities/iati-activity/result/indicator/baseline/dimension/element-exists-with-all-attributes.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0"?>
+
+<iati-activities version="xx">
+  <iati-activity> <!-- at least 1 -->
+    <iati-identifier></iati-identifier>  <!-- 1 and only 1-->
```
Comment (dalepotter) [inline tests/should-pass/iati-activities/iati-activity/result/indicator/baseline/dimension/element-exists-with-all-attributes.xml]: According to the new docs format, we need to remove all these `<!-- 1 and only 1-->` comments

File: tests/should-pass/iati-activities/iati-activity/result/indicator/baseline/dimension/element-exists-with-all-attributes.xml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <indicator measure="xx">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <baseline year="2017" value= "xx">
+          <dimension name= "xx" value="xx"><!-- dimension element included with all possible attributes for elements -->
```
Comment (dalepotter) [inline tests/should-pass/iati-activities/iati-activity/result/indicator/baseline/dimension/element-exists-with-all-attributes.xml]: Comment should be prepended by `PASS:`

File: tests/should-pass/iati-activities/iati-activity/result/indicator/baseline/dimension/element-exists-with-all-attributes.xml (hunk 2 above)
Comment (dalepotter) [reply tests/should-pass/iati-activities/iati-activity/result/indicator/baseline/dimension/element-exists-with-all-attributes.xml]: Same for other test case files.

File: tests/should-pass/iati-activities/iati-activity/result/indicator/baseline/dimension/element-exists-with-all-attributes.xml (hunk 3 above)
Comment (dalepotter) [reply tests/should-pass/iati-activities/iati-activity/result/indicator/baseline/dimension/element-exists-with-all-attributes.xml]: Same for other test case files.

Comment (dalepotter) [reply tests/should-pass/iati-activities/iati-activity/result/indicator/baseline/dimension/element-exists-with-all-attributes.xml]: Suggest `possible attributes` -> `optional attributes`.

File: tests/should-pass/iati-activities/iati-activity/result/indicator/baseline/dimension/element-exists-with-value-attribute.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0"?>
```
Comment (dalepotter) [inline tests/should-pass/iati-activities/iati-activity/result/indicator/baseline/dimension/element-exists-with-value-attribute.xml]: Suggest this test is moved to `...iati-activity/result/indicator/baseline/dimension/_value/element-exists-with-value-attribute.xml`

File: tests/should-pass/iati-activities/iati-activity/result/indicator/baseline/dimension/element-exists-with-name-attribute.xml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0"?>
```
Comment (dalepotter) [inline tests/should-pass/iati-activities/iati-activity/result/indicator/baseline/dimension/element-exists-with-name-attribute.xml]: Suggest this test is moved to `...iati-activity/result/indicator/baseline/dimension/_name/element-exists-with-value-attribute.xml`

**Review by dalepotter — CHANGES_REQUESTED** [review]
Need to add tests for namespace elements and attributes, for example as done for changes to `iati-activity/result/reference` [here](https://github.com/IATI/IATI-Schemas/blob/3cae387909fb2987d89e1e12b3462185de80b8b2/tests/should-pass/iati-activities/iati-activity/result/reference/reference-element-present-with-namespace-attribute.xml#L25) for an attribute and [here](https://github.com/IATI/IATI-Schemas/blob/3cae387909fb2987d89e1e12b3462185de80b8b2/tests/should-pass/iati-activities/iati-activity/result/reference/reference-element-present-with-namespace-element.xml#L25-L27) for an element.

(NOTE: You need to add the namespace declaration to `iati-activity` for this to work - [example here](https://github.com/IATI/IATI-Schemas/blob/3cae387909fb2987d89e1e12b3462185de80b8b2/tests/should-pass/iati-activities/iati-activity/result/reference/reference-element-present-with-namespace-attribute.xml#L3))

Suggested test cases here would be:

```
<baseline year="2017" value= "xx">
   <dimension name= "xx" value="xx" example-namespace:namespace-attribute="xx"/><!-- PASS: ADD COMMENT HERE :-) -->
<baseline>
```

and 

```
<baseline year="2017" value= "xx">
   <dimension name= "xx" value="xx">
      <example-namespace:namespace-element></example-namespace:namespace-element><!-- PASS: ADD COMMENT HERE :-) -->
   </dimension>
<baseline>
```

**Review by dalepotter — APPROVED** [review]
Nice!

**Pull request merged**